Makefile.am

```diff
@@ -212,6 +204,13 @@ tools/debian/copyright: tools/debian/copyright.template $(MANIFESTS)
 	$(AM_V_GEN) $(srcdir)/tools/build-debian-copyright > $@.tmp
 	$(MV) $@.tmp $@
 
+distdir:
```

Comment:
Hmm, do we now need to use `distdir:` and that recursive make exclusively to run the `ENABLE_DOC` check as the first thing? TBH I'd rather pay the price of doing some unnecessary stuff before failing (how often is that going to happen anyway) and keep using the official API/hooks instead (and avoid recursively calling `make` -- this is such a trap)

Reply:
No.  This is really about the fact that the default `distdir:` rule depends (inexplicably) on `$(BUILT_SOURCES)`.  We could come up with ways to get stuff out of `BUILT_SOURCES` (cf. #16025 and some even more exotic efforts for gdbus-codegen) or we could just avoid the problem entirely, which is now my preferred solution.

Reply:
We could also just decide we don't care about this.  I'm using this "clean up `make dist`" approach as a way to seek out legitimate bugs in our build system, and have already gotten quite a few of them (including the first commit of this PR, which absolutely should go in) but avoiding extra work for the sake of it is maybe not the most important thing in the world...

Reply:
Ah I see, I wasn't aware of implicit `BUILT_SOURCES` dependency. I guess it's a bit philosophical whether these should be in the dist tarball, but `gdbus-codegen` and `glib-compile-resources` (the only two cases of BUILT_SOURCES, AFAICS) are already covered by our build deps.

So, I don't veto this, and don't have a strong opinion, but this seems to introduce hacks instead of getting rid of them. There's some tension between "minimal dist" and "simple Makefile.am" here :sob:

Reply:
I agree, completely.  I'm happy to drop the second commit and only do the first one.

Might also file a bug against automake while I'm at it.  This is a strange design decision.  Probably they'd advise me that if I don't like it, I should just override the `distdir:` target :rofl:

Reply:
I filed https://debbugs.gnu.org/cgi/bugreport.cgi?bug=49317

Let's just go with the first commit here.  The extra building is silly, bit nothing to lose sleep over.

Reply:
automake seems to agree that this is a bug and proposed adding an option to fix it, and asked me to write a patch.  Cool.  I love perl.  Uh...

I feel better about the hack, to be honest, if I can add a comment pointing to an upstream commit showing that it will eventually be removed.

Reply:
The patch went upstream.  I added a comment referencing it and removed the docs stuff in order to keep it purely related to what upstream accepted.